src/app/service/authentication.service.ts

```diff
@@ -138,22 +149,27 @@ export class AuthenticationService {
    * @returns {Promise} Promise
    */
   async deleteuser(password: string) {
-    return from(this._apiClient.post(`/users/${JSON.parse(localStorage.getItem('currentUser')).id}/delete`, {'password' : password}));
+    // FIXME: es wird auch eine Erfolgsmeldung ausgegeben, wenn das Passwort falsch ist
+    // Bitte entsprechende Meldung einbauen
+    // Fehler-Response ist im OneNote
+    return from(this._apiClient.post(`/users/${JSON.parse(localStorage.getItem('currentUser')).user_id}/delete`, { 'password': password }));
   }
 
-    /**
+  /**
    * sends the new password after the reset
    * @param {string} password new password
    * @returns {Promise} Promise
    */
   async postNewPassword(password: string) {
-    await this._apiClient.request({
-      url: '/users/reset/token',
-      data: {password},
-      method: 'POST',
-    });
-  }
+    const url: string[] = this._router.routerState.snapshot.url.split('/');
```

Comment:
Sehe ich das richtig, dass die Methode nur funktioniert, wenn sie zufällig von einer Komponente aufgerufen wird, die unter` /users/reset/:user_id` liegt?

Reply:
Ja, das ist in der API so definiert. Es wird per E-Mail ein entsprechender Link mit einem Token in der URL versendet. Beim Festlegen des neuen Passwords muss dieser Token wieder mitgesendet werden. Deshalb zuerst die Prüfung und dann das Extrahieren des Token und anschließend das Senden des Requests mit Token.

Reply:
Aber es wäre doch besser, anstatt sich im AuthenticationService darauf zu verlassen, dass sich *niemals* die URL fürs Zurücksetzen des Passworts ändert, das Token als Parameter vom NewPasswordComponent mitgegeben wird.

Reply:
gefixt